### PR TITLE
TimeSeries panel: Do not reinitialize plot when thresholds mode change

### LINKED ISSUE #35848: GraphNG: Plot reinitializes unnecessarily because of thresholdStyle
**What happened**: GraphNG reconfig update after each query request in PanelQueryRunner.ts

**What you expected to happen**: only update after configuration changing, -> improvement Performance of uPlot

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0.0-beta2
- Data source type & version: 
- OS Grafana is installed on: wsl2/docker
- User OS & Browser: chrome
- Grafana plugins: TimeSeries
- Others:

PanelQueryRunner.ts function   getData
```
 getData(options: GetDataOptions): Observable<PanelData> {
......
         if (!sameStructure) {

             sameStructure = compareArrayValues(lastData, processedData.series, **compareDataFrameStructures);
        }
        if (!sameStructure) {
          for each new frame structureRev will be incremented and this results into GraphNG Reconfig
          **structureRev++;**
        }

```
### shallowCompare returns always with false in case of comparing "thresholdsStyle-key", is the last key in key-array

```
export function shallowCompare<T extends {}>(a: T, b: T, cmp: Cmp = defaultCmp) {
  if (a === b) {
    return true;
  }

  const aKeys = Object.keys(a);
  const bKeys = Object.keys(b);

  if (aKeys.length !== bKeys.length) {
    return false;
  }

  for (let key of aKeys) {

**Key: thresholdsStyle** come always with false back
    //@ts-ignore
    **if (!cmp(a[key], b[key])) {**
      return false;
    }
  }

  return true;
}

```

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/35848
Root cause and possible ways forward described in https://github.com/grafana/grafana/issues/35848#issuecomment-864002732

### CODE REVIEW

File: public/app/plugins/panel/timeseries/config.ts [hunk 1]
```diff
@@ -177,14 +179,19 @@ export function getGraphFieldConfig(cfg: GraphFieldConfig): SetFieldConfigOption
       commonOptionsBuilder.addAxisConfig(builder, cfg);
       commonOptionsBuilder.addHideFrom(builder);
 
-      builder.addSelect({
-        path: 'thresholdsStyle.mode',
```
Comment (torkelo) [inline public/app/plugins/panel/timeseries/config.ts:181]: We use nested paths in other places, what is the problem in this case?

**Comment by torkelo** [comment]
> Since the mode is defined as string, rather than object, a bew object is created (

Where is the new object created?

**Comment by dprokop** [comment]
> Where is the new object created?

@torkelo I think it's created by `set` in https://github.com/grafana/grafana/blob/83f62c191263dfb52eabb65fe7e7aa3027d8bbc0/packages/grafana-data/src/field/fieldOverrides.ts#L299 
We get mode as string from https://github.com/grafana/grafana/blob/f4a40a4d85f2f20d757296343e6d43e8bc70cd0f/packages/grafana-ui/src/components/uPlot/config.ts#L287 and set creates a new object each time for thresholdsStyle.mode path.

**Comment by torkelo** [comment]
Ah I understand now, maybe we should remove support for nested paths :)

**Comment by dprokop** [comment]
> Ah I understand now, maybe we should remove support for nested paths :)

I think so too. We don't use it that often, yet it creates issues when we do :/

File: public/app/plugins/panel/timeseries/config.ts (hunk 1 above)
Comment (dprokop) [reply public/app/plugins/panel/timeseries/config.ts:181]: see my comment below

**Review by torkelo — APPROVED** [review]

**Pull request merged**